packages/rich-text/src/plugins/Table/withInsertFragmentTable.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  ELEMENT_DEFAULT,
+  getEndPoint,
+  getPluginType,
+  getStartPoint,
+  getTEditor,
+  hasNode,
+  replaceNodeChildren,
+  select,
+  TElement,
+  withoutNormalizing,
+  WithPlatePlugin,
+} from '@udecode/plate-common';
+import { ELEMENT_TABLE, getTableAbove, getTableGridAbove } from '@udecode/plate-table';
+import cloneDeep from 'lodash/cloneDeep.js';
+import { Path } from 'slate';
+
+import { insertNodes, PlateEditor } from '../../internal';
+
+/**
+ * If inserting a table,
+ * If block above anchor is a table,
+ * - Replace each cell above by the inserted table until out of bounds.
+ * - Select the inserted cells.
+ */
+export const withInsertFragmentTable = (editor: PlateEditor, { options }: WithPlatePlugin) => {
```

Comment:
Is there a way to simplify this logic make use of the upstream function as much as possible? something like this pseudocode:

```javascript
import {withInsertFragmentTable as upstream} from '@plate/table'

export const withInsertFragmentTable = (editor, { options }) => {
	myEditor.insertFragment = (fragment) => {
	    if(condition){
         upstream(...)
      }else {
        // our logic
       }
	}
}
```

Or 

```javascript
import {withInsertFragmentTable as upstream} from '@plate/table'

export const withInsertFragmentTable = (editor, { options }) => {
	myEditor.insertFragment = (fragment) => {
	    upstream(...)

      if(condition){
        // undo upstream and override with our element
       }
	}
}
```

Reply:
I don't want to adjust the logic of this file as it's copied from plates implementation the only part that was adjusted is below

Reply:
I know it's copied from plate which is why I want to eliminate as much as possible of it to:
1. get latest fixes when we upgrade
2. have less code to maintain